var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -214,6 +214,8 @@ class PyTorch(PythonPackage, CudaPackage, ROCmPackage):
     depends_on("cuda@10.2:11.4", when="@1.10+cuda", type=("build", "link", "run"))
     depends_on("cuda@9.2:11.4", when="@1.6:1.9+cuda", type=("build", "link", "run"))
     depends_on("cuda@9:11.4", when="@:1.5+cuda", type=("build", "link", "run"))
+    # https://github.com/pytorch/pytorch/issues/122169
+    depends_on("cuda@:12.3", when="platform=aarch64:", type=("build", "link", "run"))
```

Comment:
@alalazo this constraint is also being ignore, the concretizer is choosing cuda 12.4 and it fails.